src/services/completions.ts

```diff
@@ -1776,6 +1776,18 @@ namespace ts.Completions {
             }
         }
 
+        // If the symbol is for a member of an object type and is the internal name of an ES
+        // symbol, it is not a valid entry. Internal names for ES symbols start with "__@"
+        if (symbol.flags & SymbolFlags.ClassMember) {
+            const escapedName = symbol.escapedName as string;
+            if (escapedName.length >= 3 &&

```

Comment:
There's a `isReservedMemberName` inside `checker.ts` that we should just move to utilities and reuse here.

Reply:
`isReservedMemberName` explicitly *does not* remove ES symbols, only `__call`, `__new`, etc. since those aren't actually members. ES symbol, however, *are* members. Also, `isReservedMemberName` is already executed as part of the call to `getApparentProperties` on line 287.

Reply:
I wonder if `SymbolObject.prototype.name` should return undefined for internal names, though I'm not sure if that would have other undesirable effects.

Reply:
> `isReservedMemberName` explicitly does not remove ES symbols

I misread it, OK. Would still be nice to have all these escaped name inspection snippets in one place... doesn't _really_ matter, though.

> I wonder if `SymbolObject.prototype.name` should return undefined for internal names, though I'm not sure if that would have other undesirable effects.

Ideally we (internally) should never use `.name` on an internally named symbol, since we (internally) swapped everything to `.__escapedName` before re-implementing `.name` (with proper escaping) for external compatibility. Would making it return `undefined` on internal symbol names likely break any external consumers? If no, I'm all for it.

Reply:
I might look at the `.name` thing some other time.